
<div style="text-align: center;">

  <!-- Logotipo -->
  <img src="img/LogoUesc.png" alt="UESC" style="width: 60px; display: block; margin: 0 auto 5px auto;">

  <!-- Cabeçalho -->
  <div style="font-size: 20px;"><strong>UNIVERSIDADE ESTADUAL DE SANTA CRUZ - UESC</strong></div>
  <div style="font-size: 18px;">DEPARTAMENTO DE ENGENHARIAS E COMPUTAÇÃO</div>
  <div style="font-size: 18px;">ENGENHARIA QUÍMICA</div>

  <hr style="margin: 20px 0; border: 1px solid #0074B7;">

  <!-- Título do Trabalho -->
  <div style="font-size: 22px; font-weight: bold; margin-bottom: 10px;">Introdução à Controle de Processos</div>
  <div style="font-size: 20px; font-weight: bold; margin-bottom: 10px;">Exercício 1: Separadores Trifásicos - Computacional</div>

  <!-- Autor e orientador -->
  <div style="font-size: 18px;">Prof. Dr. E.R.Edwards</div>

<hr style="margin: 20px 0; border: 1px solid #0074B7;">
  <!-- Data -->
<!-- <div style="font-size: 16px; margin-top: 15px;">Ilhéus - BA, Abril de 2025</div> -->
</div>

<div style="background-color: #e6f2ff; border-left: 6px solid #0074B7; padding: 10px; font-size: 16px;">
  <strong>Controle PI da Interface Água-Óleo em um Separador Trifásico.</strong> 
</div>

Um separador trifásico horizontal é utilizado para separar gás, óleo e água proveniente da produção de Petróleo. A pressão do gás  é mantida constante por uma malha de controle independente. Para evitar que a água ultrapasse o vertedouro e contamine a corrente de óleo, a interface água-óleo deve ser mantida próximo do valor de referência $h_{sp}=0.90 \ m$. O vertedouro interno encontra-se na altura $h_v = 1.20 \ m$. Após uma perturbação operacional, o operador registrou os valores da interface apresentados na tabela abaixo:

| Tempo (h) | Interface h(m) |
|---|-------|
| 0 | 1.02| 
| 1 | 0.99|
| 2 | 0.97|
| 3 | 0.95|
| 4 | 0.93|

A válvula de saída da água é controlada por um controlador PI com ganho proporcional $K_p = 3.5$, ganho integral $K_i = 0.8$, passo de integração $\Delta t = 1 \ h$ e condição inicial $I_o = 0$. Considere as seguintes equações:

$$ e_k = h_k - h_{sp} $$

$$  I_{k+1} = I_k + e_k. \Delta t $$

$$ u_k =  K_p. e_k + K_i.I_k  $$

<img src="img/Tanque-separacao-trifasico.png" alt="UESC" style="width: 600px; display: block; margin: 0 auto 5px auto;"><p style="text-align: center;"><strong>Figura: Tanque de Separação de água, óleo e gás proveniente do poço de Petróleo.</strong></p>

1. Calcule os valores do erro ($e_k$), da integral acumulada ($I_k$) e da ação de controle ($u_k$) para todos os instantes de tempo da tabela.

2. Com base nos resultados obtidos, explique se a válvula de saída de água tende a abrir ou fechar ao longo do tempo.

3. Explique fisicamente por que a interface aproxima-se gradualmente do valor de referência.

4. Discuta se existe risco da interface atingir o vertedouro e contaminar a corrente de óleo.

__Solução Computacional__

In [9]:
# Dados de entrada extraidos do enunciado.

# Integral inicial;
I0 = 0

# A partir dos dados da tabela, temos:
h0, h1, h2, h3, h4  = 1.02, 0.99, 0.97, 0.95, 0.93 

# Dados do enunciado:
hsp = 0.90  # metros
kp = 3.5    # Ganho proporcional
ki = 0.8   # Ganho integral - teste com 0.4 para acertar o u1=0.411
Delta_t = 1  # horas

__1 - Cálculo do erro:__

$
e_k = h_k - h_{sp} 
$

In [10]:
e0 = h0 - hsp
e1 = h1 - hsp
e2 = h2 - hsp
e3 = h3 - hsp
e4 = h4 - hsp

# Imprimindo os erros:
print('e0 = {:.3f}'.format(e0))
print('e1 = {:.3f}'.format(e1))
print('e2 = {:.3f}'.format(e2))
print('e3 = {:.3f}'.format(e3))
print('e4 = {:.3f}'.format(e4))

e0 = 0.120
e1 = 0.090
e2 = 0.070
e3 = 0.050
e4 = 0.030


2 - Cálculo da Integral acumulada.

$ I_{k+1} =  I_k + e_k.\Delta t $

In [11]:
# k = 0
I1 = I0 + e0 *  Delta_t
I2 = I1 + e1 *  Delta_t
I3 = I2 + e2 *  Delta_t
I4 = I3 + e3 *  Delta_t
I5 = I4 + e4 *  Delta_t

# Imprimindo a integral acumulada:
print('I1 = {:.3f}'.format(I1))
print('I2 = {:.3f}'.format(I2))
print('I3 = {:.3f}'.format(I3))
print('I4 = {:.3f}'.format(I4))
print('I5 = {:.3f}'.format(I5))

I1 = 0.120
I2 = 0.210
I3 = 0.280
I4 = 0.330
I5 = 0.360


3 - Cálculo da Ação de Controle:

$ u_k = K_p.e_k + K_i . I_k  $ 

In [12]:
u0 = kp * e0 + ki * I0
u1 = kp * e1 + ki * I1
u2 = kp * e2 + ki * I2
u3 = kp * e3 + ki * I3
u4 = kp * e4 + ki * I4

# Imprimindo a Ação de Controle:
print('u0 = {:.3f}'.format(u0))
print('u1 = {:.3f}'.format(u1))
print('u2 = {:.3f}'.format(u2))
print('u3 = {:.3f}'.format(u3))
print('u4 = {:.3f}'.format(u4))

u0 = 0.420
u1 = 0.411
u2 = 0.413
u3 = 0.399
u4 = 0.369


4 - Tabela Resumo:

| k | $e_k$ | $I_k$ | $u_k$ |
|---|-------|-------|-------|
| 0 | 0.12  | 0.00  | 0.420 |
| 1 | 0.09  | 0.12  | 0.411 |
| 2 | 0.07  | 0.21  | 0.413 |
| 3 | 0.05  | 0.28  | 0.399 |
| 4 | 0.03  | 0.33  | 0.369 |

__Interpretação Física__

A interface encontra-se acima do setpoint, portanto $e_k > 0$  e o controlador aumenta a ação de controle. 

- Na prática, a válvula de saída de água tende a abrir para remover mais água do separador.

- Com o aumento da vazão de saida ocorre redução do volume acumulado de água. Consequentemente, a interface aproxima-se gradualmente do valor desejado. 

__Análise do Vertedouro__

O vertedouro encontra-se em: $h_v = 1.02 \ m$. A maior interface observada foi $ 0.90 + 0.12 = 1.02 m$. Como: $1.02 < 1.20 \ m$ não existe risco imediato de transbordamento de água para a região de óleo.

O controlador esta conduzindo o sistema para uma condição operacional segura. 

__Conclusão:__

O controlador PI reduz progressivamente o erro. A válvula tende a abrir para aumentar a remoção de óleo. O vertedouro não é atingido. A operação permanece segura.









<div style="text-align: center; font-size: 12px; color: gray; margin-top: 40px;">
  Este notebook foi desenvolvido no âmbito do Grupo de Pesquisas em Modelagem Computacional da UESC.<br>
  Todos os direitos reservados © 2025
</div>